### ISSUE TITLE
Change launchMode from singleTop to singleTask

### ISSUE THREAD

**Issue by blasten** [flutter/flutter#91649]
In an Android native app, an app can deep link into another app by initializing an activity.

However, when `launchMode:singleTop` a _new_ activity is created when the current activity isn't at the top
of the target task.  This means that a new Dart entrypoint is created every time a new activity is created.

Because an activity has a 1:1 relation with a Flutter engine, there's no state synchronization between these 
two activities that would allow an app to skip certain screen. e.g. a login screen.

As far as I know state restoration doesn't deal with this problem,  correct @goderbauer ?
 
The alternative is to use `singleTask`, which limits the instantiation to just one activity.

Related PRs:
https://github.com/flutter/flutter/pull/56899
https://github.com/flutter/samples/pull/883

**Comment by blasten** [comment]
@gaaclarke 
does a shared engine also share the Dart heap memory?

**Comment by goderbauer** [comment]
> As far as I know state restoration doesn't deal with this problem, correct @goderbauer ?

Currently it doesn't AFAIK. The state we collect for state restoration also does contain a lot of UI state (e.g. what screen are we currently on, how far down are we scrolled in a scrollable). Presumably, that's not the state you want to sync here?

**Comment by gaaclarke** [comment]
> does a shared engine also share the Dart heap memory?

Yep, but there are still protections from arbitrarily sharing an object between the engines.

**Comment by GaryQian** [comment]
@blasten If this does not work and is not trivial to get around engine sharing protections, let's go ahead and disable this in AndroidManifest.xml since it wont work anyways.

**Comment by christopherfujino** [comment]
@GaryQian would it make sense to lower the priority on this one, as it doesn't seem like something we will imminently be working on?

**Comment by GaryQian** [comment]
Yeah, lowered. I know of no near term plans to address this.

**Comment by Zazo032** [comment]
@GaryQian would this provide any benefit over using `singleInstance` instead?

**Comment by sagar-tide** [comment]
This will cause a problem when you integrate an android native SDK e.g google tap and pay SDK. The SDK will launch a native screen and the screen will start your activity with specific intent action. Adhere to the Android security model, especially concerning the use of intents. Upon receiving the intent, use Activity.getCallingPackage to validate that the calling activity is actually the app you wanted it to be. 

Due to `singleTask` you will never get the packageName of the calling activity. Check this thread:
https://stackoverflow.com/a/31607292

**Comment by reynirf** [comment]
I have the same issue as @sagar-tide.
